docs/kap_proposals/kap_5_ref_types_redesign.md

```diff
@@ -0,0 +1,126 @@
+# Ref Types Redesign
+
+Redesign Kapitan Secrets and rename them as `References` or `Ref`.
+
+This will likely be a breaking change.
+
+Author: @ramaro
+
+## Proposal
+
+Rename `Secrets` into `Ref` (or `References`) to improve consistency and meaning of the backend types
+by removing the `ref` backend and introducting new backends:
+
+| Type   | Description | Encrypted? | Compiles To |
+| ------ | ----------- |        --- | ----------  |
+| gpg    | GnuPG       |        Yes | hashed tag  |
+| gkms   | Google KMS  |        Yes | hashed tag  |
+| awskms | Amazon KMS  |        Yes | hashed tag  |
+| base64 | base64      |        No  | hashed tag  |
+| plain  | plain text  |        No  | plain text  |
+
+The type value will now need to be representative of the way a reference is stored via its backend.
+
+A new `plain` backend type is introduced and will compile into revealed state instead of a hashed tag.
+
+A new `base64` backend type will store a base64 encoded value as the backend suggests (replacing the old badly named `ref` backend).
+
+
+The command line for secrets will be instead:
+
+```shell
+$ kapitan refs --write gpg:my/secret1 ...
```

Comment:
We can keep `secrets` here to make it backwards compatible for a couple of versions btw.

Reply:
We will already break the `ref` type. Perhaps we can keep the `secrets` sub command but have it warn that it was renamed to `refs`?

Reply:
Yeap sgtm ^